lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -2713,6 +2713,89 @@ def ValidateParameters(cls, hvparams):
     kvm_output = cls._GetKVMOutput(kvm_path, cls._KVMOPT_MLIST)
     validate_machine_version(hvparams, kvm_output)
 
+  @classmethod
+  def AssessParameters(cls, hvparams):
+    """Check the given parameters for uncommon/suboptimal values
+
+    This should check the passed set of parameters for suboptimal
+    values.
+
+    @type hvparams: dict
+    @param hvparams: dictionary with parameter names/value
+
+    """
+
+    warnings = []
+
+    cpu_type = hvparams[constants.HV_CPU_TYPE]
+    if not cpu_type:
+      warnings.append("cpu_type is currently unset and defaults to 'qemu64'"
+                      ", please read the gnt-instance man page on the security"
+                      " and performance implications of this parameter.")
+    elif cpu_type in ("qemu32", "qemu64", "kvm32", "kvm64"):
+      warnings.append("cpu_type is currently set to '%s'"
+                      ", please read the gnt-instance man page on the security"
+                      " and performance implications of this parameter." %
+                      cpu_type)
+    elif cpu_type == "host":
+      warnings.append("cpu_type is currently set to 'host', please make"
+                      " sure all your cluster nodes have the exact same"
+                      " CPU type to allow live migrations.")
+
+    disk_aio = hvparams[constants.HV_KVM_DISK_AIO]
+    if disk_aio == "threads":
+      warnings.append("disk_aio is set to 'threads' and should be set"
+                      " to 'native' for better disk performance.")
+
```

Comment:
On which basis is `aio=native` suggested here? Other sources[1] seem to recommend `=threads`? Maybe we can avoid suggesting non critical (and sometimes not easy to explain) parts?

[1] https://kb.blockbridge.com/technote/proxmox-aio-vs-iouring/

Reply:
@saschalucas I think you are (understandably) mixing up `disk_aio=threads` and IOThreads which is the backend implementation at the controller level. See http://kvmonz.blogspot.com/p/knowledge-disk-performance-hints-tips.html (esp. "Choosing the right AIO mode") for details, but the summary is `disk_aio` should be set to `native` and the IOThreads mentioned by Proxmox should be configured at the disk controller level.

Reply:
Thanks @apoikos for clarification. Indeed I mixed things which do not belong together. However the article you have linked, recommends `native` for disk templates `plain`, `drbd` and `blockdev`, but `threads` for `file`, `sharedfile` and maybe `rbd`?

My point here is, whether we recommend only critical and generally valid configuration settings or tuning/best practices, that sometimes depends ...

Reply:
I see your point here @saschalucas. Since this is an assessment of global/cluster-level settings, individual instances and their disk backend/template can not be taken into account. However, it would be possible to combine this check with the list of enabled disk templates. This would allow the message to be a bit more specific and reduce the chance of false alerts.

Reply:
Actually I re-read the article posted by @apoikos again. Since Ganeti uses what the article calls "raw image files", it also recommends using `native` there. It only recommends `threads` in the case of non-pre-allocated/sparse qcow2 images, which Ganeti does not use/support.

In that case I think it is pretty safe to assume `native` is the best choice here.

Reply:
> When space efficient image files are used (QCOW2 without pre-allocation, or sparse raw images) the default of io='threads' may be better suited.

The later (sparse raw image) is used by `file`, `sharedfile`and maybe in analogy by `rbd`.

Reply:
Oh, I did not even know that 🤦‍♂️

But wouldn't `rbd` be either a blockdevice or userspace access, depending on configuration?

Reply:
In analogy `rbd` images are sparse (in block terminology thin). Even LVM can be thin today (thinpool). The article just mentioned the first write of an unallocated storage block. It may be argued, that on heavy workloads (where tuning make sens) a block is written more than once and hence aio=native could be better for all writes after the first. But who knows...

Reply:
I tried to come up with a quick solution that is a "bit more intelligent" (with regards to the enabled disk templates). However, I eventually just dropped this part from the PR.

It adds the groundwork to assess hvparams and everything we require _right now_ is included (`machine_version` checks). 

We can improve from here and add more assessments in future PRs.